config.mak.uname

```diff
@@ -678,60 +678,45 @@ ifeq ($(uname_S),MINGW)
 	NATIVE_CRLF = YesPlease
 	X = .exe
 ifneq (,$(wildcard ../THIS_IS_MSYSGIT))
-	htmldir = doc/git/html/
-	prefix =
+	$(error "Building with MSys is no longer supported")
+endif
```

Comment:
This looks much better.

However, the `MSYSGIT` condition guards not against _all MSys_ setups, but only against the _msysGit_ ones. I do not want to write that condition for you, as you want to learn something, right? So: can you find the correct condition that discerns between MSys and MSYS2 and use it here in a modified form?

I'm curious, what do you think about moving this `ifneq ... $(error ...) endif` to the top of the `MINGW` clause? It could potentially make things more readable, following the "early return on error" paradigm.

In [an earlier message](https://github.com/gitgitgadget/git/issues/1319#issuecomment-1321224529), you mentioned that a couple of variables need deduplication, and I cautioned against it. Having said that, the duplicate `USE_NED_ALLOCATOR = YesPlease` looks inelegant. Again, I do not want to take your learning away from you, so: can you figure out how/where that duplicate setting was introduced? You could then remove the duplicate line in a preparatory commit (i.e. as the first of two commits, see [the documentation of the interactive rebase](https://git-scm.com/docs/git-rebase#_interactive_mode) how you can reorder patches) and describe how the duplicate line got there and that it is duplicate so we're gonna remove it.

At this stage, we need to talk about commit messages. The Git project is a bit meticulous and particular about commit messages. It has some [rambling guidelines](https://git-scm.com/docs/SubmittingPatches#describe-changes) how to write good commit messages. Personally, I much prefer [these tips](https://github.blog/2022-06-30-write-better-commits-build-better-projects/) how to write good commit messages. If you're more an example-led person, compare the commit messages of 11ad90655f8f3dd1c660776cc0820638647d8c08 and of df5218b4c30b2fa1ba0d1ada4166ad85bc83be26. It's not so much that one commit message is substantially longer. It's more that a good commit message prepares the reader well to understand the patch and its motivation.

And now we need to talk about the cover letter. As you can see e.g. [in this mail](https://lore.kernel.org/git/pull.1386.git.git.1669241489179.gitgitgadget@gmail.com/), the initial PR comment is inlined between the commit message and the diffstat. If something like this landed on the Git mailing list as cover letter, readers would probably get the impression that the contribution was a bit sloppy:

> Signed-off-by: Harshil-Jani [harshiljani2002@gmail.com](mailto:harshiljani2002@gmail.com)
>
> Thanks for taking the time to contribute to Git! Please be advised that the
> Git community does not use github.com for their contributions. Instead, we use
> a mailing list ([git@vger.kernel.org](mailto:git@vger.kernel.org)) for code submissions, code reviews, and
> bug reports. Nevertheless, you can use GitGitGadget (https://gitgitgadget.github.io/)
> to conveniently send your Pull Requests commits to our mailing list.
>
> Please read the "guidelines for contributing" linked above!

So let's edit this accordingly? The cover letter is an excellent place to introduce yourself and mention how you got interested in contributing to Git, and in contributing _this_ patch in particular.

Reply:
Thankyou for the brief. It would be a good learning experience for me by following the provided advice. Will be making this done in weekends.